immuni_exposure_ingestion/models/upload_eu.py

```diff
@@ -0,0 +1,99 @@
+#    Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#    Please refer to the AUTHORS file for more information.
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import List
+
+from bson import ObjectId
+from mongoengine import BooleanField, DateField, Document, EmbeddedDocumentListField, StringField
+from pymongo.cursor import Cursor
+
+from immuni_common.models.mongoengine.temporary_exposure_key import TemporaryExposureKey
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class UploadEu(Document):
```

Comment:
As already suggested in another comment, let's create a superclass from which both `Upload` and `UplaodEU` inherit from. The current implementation is clearly against DRY, thus unmaintainable in the long run. 🙏

Reply:
we would not want to compromise the functioning of the current flow and also there are small but significant differences

Reply:
> we would not want to compromise the functioning

IMHO, this is not a good reason to duplicate code. 😞 Proper testing and abstraction should provide enough confidence in this regard.

> small but significant differences

This is a big incentive for keeping the common part unique. One reading the new class can then focus on the small differences as they'll be the only one present in this class, instead of losing them in between the cluttered code.